src/main/java/com/zaxxer/hikari/metrics/prometheus/PrometheusMetricsTrackerFactory.java

```diff
@@ -19,29 +19,91 @@
 import com.zaxxer.hikari.metrics.IMetricsTracker;
 import com.zaxxer.hikari.metrics.MetricsTrackerFactory;
 import com.zaxxer.hikari.metrics.PoolStats;
+import io.prometheus.client.CollectorRegistry;
 
 /**
+ * <p>
+ * Use default CollectorRegistry and default HikariCPCollector:
  * <pre>{@code
  * HikariConfig config = new HikariConfig();
  * config.setMetricsTrackerFactory(new PrometheusMetricsTrackerFactory());
  * }</pre>
+ * </p>
+ * <p>
+ * Use provided CollectorRegistry and default HikariCPCollector:
+ * <pre>{@code
+ * HikariConfig config = new HikariConfig();
+ * config.setMetricsTrackerFactory(new PrometheusMetricsTrackerFactory(registry));
+ * }</pre>
+ * </p>
+ * <p>
+ * Use provided CollectorRegistry and provided HikariCPCollector:
+ * <pre>{@code
+ * HikariConfig config = new HikariConfig();
+ * config.setMetricsTrackerFactory(new PrometheusMetricsTrackerFactory(registry, collector));
+ * }</pre>
+ * </p>
  */
 public class PrometheusMetricsTrackerFactory implements MetricsTrackerFactory {
 
-   private static HikariCPCollector collector;
+   private HikariCPCollector collector;
+   private CollectorRegistry registry;
+
+   private static volatile HikariCPCollector globalCollector;
+
+   /**
+    * If nothing is provided, use {@link CollectorRegistry.defaultRegistry} and {@link globalCollector} as default
+    */
+   public PrometheusMetricsTrackerFactory() {
+      this(CollectorRegistry.defaultRegistry);
+   }
+
+   /**
+    * If a CollectorRegistry is provided, use provided CollectorRegistry and {@link globalCollector} as default
+    */
+   public PrometheusMetricsTrackerFactory(CollectorRegistry registry) {
+      if (globalCollector == null) {
+         synchronized (PrometheusMetricsTrackerFactory.class) {
+            if (globalCollector == null) {
+               globalCollector = new HikariCPCollector();
+            }
+         }
+      }
+      this.registry = registry;
+      this.collector = globalCollector.register(registry);
+   }
+
+   /**
+    * If both a CollectorRegistry and a Collector are provided, use provided CollectorRegistry and Collector
+    */
+   public PrometheusMetricsTrackerFactory(CollectorRegistry registry, HikariCPCollector collector) {
+      this.registry = registry;
+      this.collector = collector.register(registry);
+   }
```

Comment:
Why is this constructor necessary?  In particular, if ``HikariCPCollector`` is internal then this constructor is never used.